### ISSUE TITLE
Embedded linux system：Verifying kprobe... ERR:-22

### ISSUE THREAD

**Issue by unintialized** [wkz/ply#32]
hi, 
I got an error when I tried to run ply ，My system is an embedded development board, the model is TI am5748 IDK ，It is an arm A15 (armv7l) architecture cpu。

error message:

```
root@am57xx-evm:~# ply -T
Verifying kernel config (/proc/config.gz)... OK
Ensuring that debugfs is mounted... OK
Verifying kprobe... ERR:-22
ERROR
Verifying tracepoint... /bin/sh: line 79:  1203 Aborted                 (core dumped) $PLYBIN 'tracepoint:sched/sched_switch { exit(0); }' 2> /dev/null
ERROR


root@am57xx-evm:~# ply -dS "tracepoint:sched/sched_process_exit { ${1} }"
ply: provider/tracepoint.c:197: tracepoint_parse: Assertion `offs == type_offsetof(t, t->sou.fields[n - 1].name)' failed.
Aborted (core dumped)


root@am57xx-evm:~# cat test.ply
#!/usr/bin/env ply
kprobe:do_sys_open
{
            printf("%v(%v): %s\n",
               comm, pid, str(arg1));
}
root@am57xx-evm:~# ./test.ply
error: unable to load kprobe:do_sys_open, errno:22
error: no output from kernel bpf verifier, retry with debugging enabled
ERR:-22



root@am57xx-evm:~# zgrep "BPF" /proc/config.gz
# CONFIG_CGROUP_BPF is not set
CONFIG_BPF=y
CONFIG_BPF_SYSCALL=y
# CONFIG_BPF_JIT_ALWAYS_ON is not set
# CONFIG_NETFILTER_XT_MATCH_BPF is not set
# CONFIG_BPFILTER is not set
CONFIG_NET_CLS_BPF=y
# CONFIG_NET_ACT_BPF is not set
CONFIG_BPF_JIT=y
# CONFIG_BPF_STREAM_PARSER is not set
CONFIG_HAVE_EBPF_JIT=y
# CONFIG_NBPFAXI_DMA is not set
CONFIG_BPF_EVENTS=y
# CONFIG_TEST_BPF is not set

root@am57xx-evm:~# zgrep "KPROBE" /proc/config.gz
CONFIG_KPROBES=y
CONFIG_HAVE_KPROBES=y
CONFIG_KPROBE_EVENTS=y
# CONFIG_KPROBES_SANITY_TEST is not set
# CONFIG_ARM_KPROBES_TEST is not set

root@am57xx-evm:~# zgrep "FTRACE" /proc/config.gz
# CONFIG_PSTORE_FTRACE is not set
CONFIG_HAVE_DYNAMIC_FTRACE=y
CONFIG_HAVE_DYNAMIC_FTRACE_WITH_REGS=y
CONFIG_HAVE_FTRACE_MCOUNT_RECORD=y
CONFIG_FTRACE=y
# CONFIG_FTRACE_SYSCALLS is not set
CONFIG_DYNAMIC_FTRACE=y
CONFIG_DYNAMIC_FTRACE_WITH_REGS=y
CONFIG_FTRACE_MCOUNT_RECORD=y
# CONFIG_FTRACE_STARTUP_TEST is not set


root@am57xx-evm:~# cat /sys/kernel/debug/tracing/events/sched/sched_process_exit/format
name: sched_process_exit
ID: 56
format:
        field:unsigned short common_type;       offset:0;       size:2; signed:0;
        field:unsigned char common_flags;       offset:2;       size:1; signed:0;
        field:unsigned char common_preempt_count;       offset:3;       size:1; signed:0;
        field:int common_pid;   offset:4;       size:4; signed:1;
        field:unsigned short common_migrate_disable;    offset:8;       size:2; signed:0;
        field:unsigned short common_padding;    offset:10;      size:2; signed:0;

        field:char comm[16];    offset:16;      size:16;        signed:0;
        field:pid_t pid;        offset:32;      size:4; signed:1;
        field:int prio; offset:36;      size:4; signed:1;

print fmt: "comm=%s pid=%d prio=%d", REC->comm, REC->pid, REC->prio


```
My linux is ：PREEMTP_RT 
```
root@am57xx-evm:~# uname -a
Linux am57xx-evm 4.19.94-rt39-ga242ccf3f1 #2 SMP PREEMPT RT Sun May 30 17:58:43 PDT 2021 armv7l GNU/Linux

root@am57xx-evm:~# ply -v
ply  (linux-version:267067~4.19.59)

```

What should I do？

Thanks You,
unintialized

**Comment by wkz** [comment]
I would start by trying out the same thing in ply's CI environment (`make arm-check-install-tc` from the `test/` directory). If it works there, I would start trying to diff the two setups (e.g. using `strace`).

I know that there are bugs around `struct` alignment, likely that is cause of the assertion failure. But that does not explain why `ply -T` fails.

**Comment by wkz** [comment]
The issue around `struct` alignment should now have been fixed in 6e25f69. Which issues remain on the latest version of ply?

**Comment by woniu2014** [comment]
system is an embedded development board ,I use ply 2.2.0
error: unable to load kprobe:do_sys_open, errno:22
error: no output from kernel bpf verifier, retry with debugging enabled
ERR:-22
